pkg/envoy/lds/rbac_test.go

```diff
@@ -32,10 +32,10 @@ func TestBuildRBACPolicyFromTrafficTarget(t *testing.T) {
 			name: "traffic target without TCP routes",
 			trafficTarget: trafficpolicy.TrafficTargetWithRoutes{
 				Name:        "ns-1/test-1",
-				Destination: identity.ServiceIdentity("sa-1.ns-1.cluster.local"),
+				Destination: identity.NewFromServiceAccountString("sa-1.ns-1.cluster.local"),
```

Comment:
The whole purpose of #2218 was to not leak k8s service-account type into XDS verticals such as `rbac.go`. Why is this being changed?

Reply:
@shashankram The question you ask is valid and it seems that we are thinking about this differently.  Which is a good thing, but also we'll need to create a design doc to understand where we should go next.

Based on the SMI policies, which reference `kind: ServiceAccount`, which is clearly pointing to a Kubernetes Service Account, we can't necessarily remove the notion of Service Account.

`ServiceIdentity` makes sense in general, but in some cases (all cases right now) it is **derived** from a Kubernetes Service Account.  This is evident even in the string itself.

I should also mention that with these PRs (they are draft for that reason) I am exploring to understand what needs to happen and how that needs to happen to migrate from `K8sServiceAccount` to `ServiceIdentity`.

One option, which I'm starting to like, is to make `ServiceIdentity` be a superset of `K8sServiceAccount`.
Something like this:
```go
type ServiceIdentity struct {
	kind ServiceIdentityKind

	// serviceAccount is only used when the kind of ServiceIdentity is KubernetesServiceAccount
	serviceAccount K8sServiceAccount
	trustDomain    string
}
```

This will (in the short term at least) match how SMI is constructed.

Reply:
That's great @draychev, this is very much needed because as you mentioned, we need to fill the dots between `ServiceIdentity` (fully qualified) and `K8sServiceAccount`.